torch/csrc/jit/codegen/cuda/kernel.cpp

```diff
@@ -349,6 +349,36 @@ void Kernel::registerExpr(Expr* expr) {
   Fusion::registerExpr(expr);
 }
 
+namespace {
+
+class KernelExprVisitor : private kir::IrVisitor {
+ public:
+  static std::vector<Expr*> getAllExprs(const Kernel* kernel) {
+    KernelExprVisitor visitor(kernel);
+    return visitor.all_exprs_;
+  }
+
+ private:
+  KernelExprVisitor(const Kernel* kernel) {
+    handle(kernel->topLevelExprs());
+  }
+
+  using kir::IrVisitor::handle;
+
+  void handle(Expr* expr) final {
+    all_exprs_.push_back(expr);
+    kir::IrVisitor::handle(expr);
+  }
+
+ private:
+  std::vector<Expr*> all_exprs_;
+};
+
+} // namespace
+std::vector<Expr*> Kernel::exprs() const {
+  return KernelExprVisitor::getAllExprs(this);
```

Comment:
It's a little more aggressive where flatten will remove all nested scopes like `IfThenElse` and `ForLoop` this will also put those in the flattened scope, it's looking to have a simple vector with every expression in `topLevelExprs()`. For now I pushed this to the tests since that's the only place it's used. If we want it in the lowering infrastructure we can revisit, move it back, and consolidate with ExprFlattener.